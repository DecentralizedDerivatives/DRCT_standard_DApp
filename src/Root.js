import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './reducers';

const middlewares = [thunk];
let composeEnhancer = compose;

if (process.env.NODE_ENV === 'development') {
  middlewares.push(createLogger());
  composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const Root = ({ children, initialState = {} }) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancer(applyMiddleware(...middlewares))
  );

  return <Provider store={store}>{children}</Provider>;
};

export default Root;
