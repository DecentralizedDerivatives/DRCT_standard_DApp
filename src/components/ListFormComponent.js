import React from 'react';
import { Field } from 'redux-form';
import { InputGroup } from 'reactstrap';
import InputNumber from './InputNumber';

export const ListFormComponent = ({
  handleSubmit,
  onSubmit,
  token,
  tokenAmount
}) => {
  return (
      <form autoComplete="off" onSubmit={handleSubmit(onSubmit)}>
        <label style={{margin: '6px 0'}}><strong>Token:</strong></label>
        <div>{token}</div>
        <br />
        <label style={{margin: '6px 6px 0 0'}}><strong>Amount:</strong></label>
        <span>{tokenAmount}</span>

        <InputGroup className="order-input-wrapper">
          <Field
            name="price"
            label="Enter Price in ETH"
            className="order-input"
            component={InputNumber}
            addonLabel="ETH"
            step="0.00001"
          />
        </InputGroup>
        <div>
          <button
            type="submit"
            className="order-submit-btn btn btn-primary"
          >
            Submit
          </button>
        </div>
      </form>
  );
};

export default ListFormComponent;
