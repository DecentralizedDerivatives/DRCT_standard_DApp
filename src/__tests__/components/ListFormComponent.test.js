import React from 'react';
import { shallow } from 'enzyme';
import { initStore } from '../../Root';
import ListFormComponent from '../../components/ListFormComponent';

describe('<ListFormComponent />', () => {
  describe('render()', () => {
    it('renders the component', () => {
      const handleSubmit = jest.fn();
      const wrapper = shallow(
        <ListFormComponent store={initStore()} handleSubmit={handleSubmit} />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
