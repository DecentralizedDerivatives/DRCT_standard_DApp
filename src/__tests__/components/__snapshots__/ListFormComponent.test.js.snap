// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListFormComponent /> render() renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListFormComponent
    handleSubmit={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "autoComplete": "off",
      "children": Array [
        <label
          style={
            Object {
              "margin": "6px 0",
            }
          }
        >
          <strong>
            Token:
          </strong>
        </label>,
        <div />,
        <br />,
        <label
          style={
            Object {
              "margin": "6px 6px 0 0",
            }
          }
        >
          <strong>
            Amount:
          </strong>
        </label>,
        <span />,
        <InputGroup
          className="order-input-wrapper"
          tag="div"
        >
          <Field
            addonLabel="ETH"
            className="order-input"
            component={[Function]}
            label="Enter Price in ETH"
            name="price"
            step="0.00001"
          />
        </InputGroup>,
        <div>
          <button
            className="order-submit-btn btn btn-primary"
            type="submit"
          >
            Submit
          </button>
        </div>,
      ],
      "onSubmit": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <strong>
            Token:
          </strong>,
          "style": Object {
            "margin": "6px 0",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Token:",
          },
          "ref": null,
          "rendered": "Token:",
          "type": "strong",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <strong>
            Amount:
          </strong>,
          "style": Object {
            "margin": "6px 6px 0 0",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Amount:",
          },
          "ref": null,
          "rendered": "Amount:",
          "type": "strong",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Field
            addonLabel="ETH"
            className="order-input"
            component={[Function]}
            label="Enter Price in ETH"
            name="price"
            step="0.00001"
          />,
          "className": "order-input-wrapper",
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "addonLabel": "ETH",
            "className": "order-input",
            "component": [Function],
            "label": "Enter Price in ETH",
            "name": "price",
            "step": "0.00001",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="order-submit-btn btn btn-primary"
            type="submit"
          >
            Submit
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "order-submit-btn btn btn-primary",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoComplete": "off",
        "children": Array [
          <label
            style={
              Object {
                "margin": "6px 0",
              }
            }
          >
            <strong>
              Token:
            </strong>
          </label>,
          <div />,
          <br />,
          <label
            style={
              Object {
                "margin": "6px 6px 0 0",
              }
            }
          >
            <strong>
              Amount:
            </strong>
          </label>,
          <span />,
          <InputGroup
            className="order-input-wrapper"
            tag="div"
          >
            <Field
              addonLabel="ETH"
              className="order-input"
              component={[Function]}
              label="Enter Price in ETH"
              name="price"
              step="0.00001"
            />
          </InputGroup>,
          <div>
            <button
              className="order-submit-btn btn btn-primary"
              type="submit"
            >
              Submit
            </button>
          </div>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Token:
            </strong>,
            "style": Object {
              "margin": "6px 0",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Token:",
            },
            "ref": null,
            "rendered": "Token:",
            "type": "strong",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <strong>
              Amount:
            </strong>,
            "style": Object {
              "margin": "6px 6px 0 0",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Amount:",
            },
            "ref": null,
            "rendered": "Amount:",
            "type": "strong",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Field
              addonLabel="ETH"
              className="order-input"
              component={[Function]}
              label="Enter Price in ETH"
              name="price"
              step="0.00001"
            />,
            "className": "order-input-wrapper",
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "addonLabel": "ETH",
              "className": "order-input",
              "component": [Function],
              "label": "Enter Price in ETH",
              "name": "price",
              "step": "0.00001",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="order-submit-btn btn btn-primary"
              type="submit"
            >
              Submit
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "order-submit-btn btn btn-primary",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
